feature-detects/audio/preload.js

```diff
@@ -1,13 +1,76 @@
 /*!
 {
-  "name": "Audio Preload Attribute",
+  "name": "Audio Preload",
   "property": "audiopreload",
-  "tags": ["audio", "media"]
+  "tags": ["audio", "media"],
+  "async" : true,
+  "warnings": ["This test is very large – only include it if you absolutely need it"],
+  "knownBugs": ["Not consistent in Safari and IE when run in Cow tests"]
 }
 !*/
 /* DOC
 Detects if audio can be downloaded in the background before it starts playing in the `<audio>` element
 */
-define(['Modernizr', 'createElement'], function(Modernizr, createElement) {
-  Modernizr.addTest('audiopreload', 'preload' in createElement('audio'));
+define(['Modernizr', 'addTest', 'docElement', 'createElement', 'test/audio'], function(Modernizr, addTest, docElement, createElement) {
+
+  Modernizr.addAsyncTest(function() {
+    var timeout;
+    var waitTime = 300;
+    var elem = createElement('audio');
+    var elemStyle = elem.style;
+
+    function testpreload(arg) {
+      clearTimeout(timeout);
+      var result = arg !== undefined && arg.type === 'loadeddata' ? true : false; //to avoid arg being 'undefined' and arg.type throwing out an error, see below comment
```

Comment:
I'll have a look but won't that throw 'undefined' if the function is triggered on timeout (iOS)?